### DISCUSSION TITLE
Problem loading project

### DISCUSSION THREAD

**Discussion by andredev93** [mia-platform/community#235] [Q&A] [answered]
### Which section?

Console

### Which is the version?

10.7

### Description

Hi!

I'm getting an error while opening the project page on an On-prem installation of the console with Github integration.

From the homepage we have multiple projects but we're getting the error below only on a few ones while others are loading correctly.

![image](https://user-images.githubusercontent.com/10067183/233099917-a1d259bb-431c-483b-b36d-fd7ba52e7b72.png)

In the inspector I see this error:
`{"statusCode":500,"code":"UNABLE_TO_FIND_TAGS","error":"Internal Server Error","message":""}`

While on backend we see errors like these:
`{"level":50,"time":1681913143598,"pid":8,"hostname":"backend-7df77ccc7c-2wcwf","reqId":"5b02bf6a-9ad9-4651-9858-a62667cc6f25","err":{"type":"RequestError","message":"Not Found","stack":"HttpError: Not Found\n    at /home/node/app/node_modules/@octokit/request/dist-node/index.js:86:21\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","name":"HttpError","status":404,"response":{"url":"https://github.*****/branches","status":404,"headers":{"access-control-allow-origin":"*","access-control-expose-headers":"ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset","connection":"close","content-encoding":"gzip","content-security-policy":"default-src 'none'","content-type":"application/json; charset=utf-8","date":"Wed, 19 Apr 2023 14:05:43 GMT","referrer-policy":"origin-when-cross-origin, strict-origin-when-cross-origin","strict-transport-security":"max-age=15724800; includeSubDomains","transfer-encoding":"chunked","x-accepted-oauth-scopes":"","x-content-type-options":"nosniff","x-frame-options":"deny","x-github-enterprise-version":"3.7.8","x-github-media-type":"github.v3; format=json","x-github-request-id":"a2804b41-6271-4f47-b860-8301355ebf76","x-oauth-scopes":"admin:enterprise, admin:gpg_key, admin:org, admin:org_hook, admin:pre_receive_hook, admin:public_key, admin:repo_hook, delete_repo, gist, notifications, repo, user, write:discussion","x-ratelimit-limit":"10000","x-ratelimit-remaining":"9980","x-ratelimit-reset":"1681914587","x-ratelimit-resource":"core","x-ratelimit-used":"20","x-runtime-rack":"0.044614","x-xss-protection":"0"},"data":{"message":"Not Found","documentation_url":"https://docs.github.com/enterprise-server@3.7/rest/reference/repos#list-branches"}},"request":{"method":"GET","url":"https://github.*****/branches","headers":{"accept":"application/vnd.github.v3+json","user-agent":"octokit-rest.js/18.12.0 octokit-core.js/3.5.1 Node.js/18.14.0 (linux; x64)","authorization":"token [REDACTED]"},"request":{}}},"msg":"fails to get branches"}`

-----

`{"level":50,"time":1681912472949,"pid":8,"hostname":"backend-7df77ccc7c-2wcwf","reqId":"7ad6b119-c778-4617-9e3d-99b5eb2a73ad","err":{"type":"RequestError","message":"Not Found","stack":"HttpError: Not Found\n    at /home/node/app/node_modules/@octokit/request/dist-node/index.js:86:21\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)","name":"HttpError","status":404,"response":{"url":"https://github.****/tags","status":404,"headers":{"access-control-allow-origin":"*","access-control-expose-headers":"ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset","connection":"close","content-encoding":"gzip","content-security-policy":"default-src 'none'","content-type":"application/json; charset=utf-8","date":"Wed, 19 Apr 2023 13:54:32 GMT","referrer-policy":"origin-when-cross-origin, strict-origin-when-cross-origin","strict-transport-security":"max-age=15724800; includeSubDomains","transfer-encoding":"chunked","x-accepted-oauth-scopes":"","x-content-type-options":"nosniff","x-frame-options":"deny","x-github-enterprise-version":"3.7.8","x-github-media-type":"github.v3; format=json","x-github-request-id":"5a5d51f4-fad3-474f-94a7-d7e7b2233f14","x-oauth-scopes":"admin:enterprise, admin:gpg_key, admin:org, admin:org_hook, admin:pre_receive_hook, admin:public_key, admin:repo_hook, delete_repo, gist, notifications, repo, user, write:discussion","x-ratelimit-limit":"10000","x-ratelimit-remaining":"9992","x-ratelimit-reset":"1681914587","x-ratelimit-resource":"core","x-ratelimit-used":"8","x-runtime-rack":"0.035018","x-xss-protection":"0"},"data":{"message":"Not Found","documentation_url":"https://docs.github.com/enterprise-server@3.7/rest/reference/repos#list-repository-tags"}},"request":{"method":"GET","url":"https://github.*******/tags","headers":{"accept":"application/vnd.github.v3+json","user-agent":"octokit-rest.js/18.12.0 octokit-core.js/3.5.1 Node.js/18.14.0 (linux; x64)","authorization":"token [REDACTED]"},"request":{}}},"msg":"fails to get tags"}`

### Actual Outcome

Error on project loading.

### Expected Outcome

Project page loading.

**Comment by davidebianchi** [comment]
Hi @andredev93! It is possible that the provider does not have access to the selected project, or the `configurationGitPath` field of the is wrong?
Because it returns a Not Found error, so I think it could be an error of this type.

**Reply by andredev93** [reply]
Hi @davidebianchi.
Github has access to the projects and `configurationGitPath` is set correctly to the configurations repository for the affected projects.
These projects were working and we even created some branches/tags from the console in the previous days.

**Reply by davidebianchi** [reply]
Is it possible that the access token is expired?

**Reply by andredev93** [reply, marked as answer]
It looks like the configurations repositories, of the affected projects, were set to **Private visibility** on Github for other tests and this caused the issue. After setting them to **Internal**, the projects are correctly loaded in the console.
The repos of the other working projects had **Public** visibility as it has correctly been done by the console project creation.

**Reply by fredmaggiowski** [reply]
If you need them to be private you can keep them private, just remember to give the service account proper authorization
